src/coldfront_plugin_api/urls.py

```diff
@@ -10,16 +11,68 @@
 
 
 class AllocationViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    This viewset implements the API to Coldfront's allocation object
+    The API allows filtering allocations by any of Coldfront's allocation model attributes,
+    as well as by Allocation Attributes.
+
+    Filtering by allocation model attributes
+    (i.e by pk, start_date, or even by attributes that span relationships, like project PI email)
+    is possible by using the query parameters directly as Django query filters.
+
+    For example, to filter by the allocation's project PI email: "/api/allocations?project__pi__email=test@bu.edu"
+
+    Filtering using this method requires knowledge of the Allocation model schema and Django's filters
+    Documentation of Coldfront Allocation model: https://coldfront.readthedocs.io/en/latest/apidocs/allocations/
+    Documentation for Django's query syntax: https://docs.djangoproject.com/en/5.0/topics/db/queries/#lookups-that-span-relationships
+
+    To filter by allocation attribute (AA) (i.e Quota Limits CPU),
+    name the query as the AA (case insensitive), prefix with "attr_", and replace all space with "_"
+
+    For all filters, you can also filter with conditional OR,
+    by providing multiple values seperated by comma
+
+    For example, to find allocations with AA Quota Limits CPU equal to 2 OR 4:
+    "/api/allocations?attr_quota_limits_cpu=2,4"
+    """
+
     serializer_class = serializers.AllocationSerializer
     authentication_classes = auth.AUTHENTICATION_CLASSES
     permission_classes = [IsAdminUser]
 
     def get_queryset(self):
         queryset = Allocation.objects.filter(status__name="Active")
+        query_params = self.request.query_params
 
-        if self.request.query_params.get("all") == "true":
+        if query_params.get("all") == "true":
             queryset = Allocation.objects.all()
 
+        allocation_attr_prefix = "attr_"
+        query_split_char = ","
+
+        for query, val in query_params.items():
+            if query == "all":
```

Comment:
Why is this here?

Reply:
Because the query parameter `all` is already used here to check if we should filter against active or all allocations. 
I suppose I could use the `all` query in the for loop, but I thought that would look a bit disorganized/unclear